spec/system/consumer/shopping/products_spec.rb

```diff
@@ -157,6 +157,7 @@ def expect_product_description_html_to_be_displayed(product, html, not_include =
     expect(page).to have_selector '.reveal-modal'
     modal_should_be_open_for product
     within(".reveal-modal") do
+      expect(find_link('external site')[:target]).to eq('_blank') if href
       expect(html).to include(html)
       expect(html).not_to include(not_include) if not_include
     end
```

Comment:
Why not split this up? There are two distinct chunks of tests here. One is about the product list and the other about the modal. Boolean flags are often a code smell as well. There are some strange things in this method. This line is self-fulfilling:
```rb
      expect(html).to include(html)
```

This line also seems unnecessary:
```rb
    expect(page).to have_selector '.reveal-modal'
```

The next line should test the same thing.

Reply:
I have removed those redundant assertions.
I tried to make the other assertions better but have not found a clear way to do so. I think the intention is to check whether in the rendered HTML the html passed into the method exists but  I couldn't come up with a solution that can make the assertion.
Also the spec right after this,
```
       it "does not show unsecure HTML" do
        product.description = "<script>alert('Dangerous!');</script><p>Safe</p>"
        product.save!

        visit shop_path
        expect(page).to have_content product.name

        expect_product_description_html_to_be_displayed(
          product, "<p>Safe</p>", "<script>alert('Dangerous!');</script>", truncate: false
        )
      end
```
says that unsecure html should not be displayed but is calling the same method at the end, I don't know what the actual behaviour should be but what's currently happening is that the unsecure HTML does show up (as text) in the page.